### ISSUE TITLE
When are Falsy values evaluated as None?

### ISSUE THREAD

**Issue by Healthedata1** [nazrulworld/fhir.resources#74]
* fhir.resources version:  R4
* Python version: 3.7.3
* Operating System: Mac OS

### Description

I am creating a FHIR object using the models and a spreadsheet as a source
When there is repeating element and their is no data in the source, my code creates an empty list.
This empty list is preserved unlike when the element is None.  I think when an element is evaluated to these Falsy types listed below I think it should behave the same ... namely the element value should be `None`. Note that in FHIR "when present, elements *cannot* be empty - they **SHALL**  have a value attribute, child elements, or extensions".

1. None
1. [] - an empty list
1. {} - an empty dict
1. () - an empty tuple
1. '' - an empty str
1. b'' - an empty bytes
1. set() - an empty set. 

### What I Did
~~~python
cs = construct_fhir_element('CapabilityStatement',dict(
id = meta.id,
url = f'{canon}CapabilityStatement/{meta.id}',
version = meta.version,
# more elements
instantiates = [i for i in meta.instantiates.split(',') if i],  #<<<< this evaluates to '[]'  instead of None 
))


print(cs.json(indent=2))
~~~

------------------------------
~~~json
...
  "kind": "requirements",
  "instantiates": [
    "foo"
  ],
  "_instantiates": [
    {
      "extension": [
        {
          "url": "http://hl7.org/fhir/StructureDefinition/capabilitystatement-expectation",
          "valueCode": "SHALL"
        }
      ]
    }
  ],
  "imports": [],
  "fhirVersion": "4.0.1",
....
~~~

I think this should be the default behavior of the classes as a convenience.

**Comment by nazrulworld** [comment]
I fully agree with you, that an empty list should be ignored just a null value.

**Comment by Healthedata1** [comment]
Although I can see the argument for persisting an empty list when building an instance.  You might want to append items in a subsequent step.  So it would not always be desirable behavior.  Maybe the class could preserve the list, but the methods would "scrub" them or not based on a parameter?

**Comment by nazrulworld** [comment]
All are default behaviors from the ``Pydnatic`` but It is possible to change that empty list, dict to None during serialization. I will work on it.

**Comment by Healthedata1** [comment]
Thanks!
